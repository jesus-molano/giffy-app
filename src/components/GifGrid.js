import React from 'react';
import useFetchGifs from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {
	const { gifs, loading } = useFetchGifs(category);
	return (
		<div className='gifs animate__animated animate__fadeIn'>
			<h3>{category}</h3>
			{loading && <p className="animate__animated animate__flash">Loading...</p>}
			<div className='card-grid animate__animated animate__fadeIn'>
				{gifs.map(({ id, title, url }) => (
					<GifGridItem key={id} title={title} url={url} />
				))}
			</div>
		</div>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
}

export default GifGrid;
