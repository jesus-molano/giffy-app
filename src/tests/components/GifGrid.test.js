import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
// import useFetchGifs from '../../hooks/useFetchGifs';
// jest.mock('../../hooks/useFetchGifs');

describe('GifGrid component tests', () => {
  const category = 'One Punch';

  test('should show the component correctly', () => {
    // useFetchGifs.mockReturnValue({
    //   data: [],
    //   loading: true,
    // });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  })
  test('should show items when the imgs are loaded', () => {

    // const gifs = [{
    //   id: 'ABC',
    //   url: 'https://hola/hola.jpg',
    //   title: 'Cualquier Cosa',
    // }]
    // useFetchGifs.mockReturnValue({
    //   data: gifs,
    //   loading: false,
    // });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  })
  
})
