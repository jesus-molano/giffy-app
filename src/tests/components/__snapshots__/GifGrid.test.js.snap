// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid component tests should show items when the imgs are loaded 1`] = `
<div
  className="gifs animate__animated animate__fadeIn"
>
  <h3>
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</div>
`;

exports[`GifGrid component tests should show the component correctly 1`] = `
<div
  className="gifs animate__animated animate__fadeIn"
>
  <h3>
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</div>
`;
